### PR TITLE
Use '/' instead of '\' in lockfile

### PR DESCRIPTION
**Pull request by ChengTian**
parent #1286

### CODE REVIEW

File: src/Microsoft.Framework.Runtime/DependencyManagement/LockFileFormat.cs [hunk 1]
```diff
@@ -243,7 +243,7 @@ private IPackageFile ReadPackageFile(string property, JToken json)
         private JProperty WritePackageFile(IPackageFile item)
         {
             var json = new JObject();
-            return new JProperty(item.Path, new JObject());
+            return new JProperty(item.Path.Replace('\\', '/'), new JObject());
```
Comment (troydai) [inline src/Microsoft.Framework.Runtime/DependencyManagement/LockFileFormat.cs]: nit: we have at least two places have codes to unify the separator. should be have common code?

Comment (ChengTian) [inline src/Microsoft.Framework.Runtime/DependencyManagement/LockFileFormat.cs]: We do have several places using this
untitled

I can add a method to `NuGet.PathUtility`. Does `PathUtility.GetForwardSlashSeparatedPath(string path)` sound good?

Comment (davidfowl) [inline src/Microsoft.Framework.Runtime/DependencyManagement/LockFileFormat.cs]: GetPathWithForwardSlashes

**Comment by davidfowl** [comment]
This will probably break things in the NuGet resolver. Make sure the translation to physical paths works on all OSes

**Comment by ChengTian** [comment]
Convert path separators back to `Path.DirectorySeparator` when reading lockfile. This is the easiest way to resolve introduced bugs. Need to check startup performance impact.

**Comment by ChengTian** [comment]
Verified with dotTrace on MusicStore project. This change doesn't have observable impact on performance of lockfile reading.

**Comment by davidfowl** [comment]
Is this good to go ?

**Comment by ChengTian** [comment]
Yeah, just rebased and did `build verify`. Please sign off.

**Comment by troydai** [comment]
:shipit: after rebase.

**Comment by ChengTian** [comment]
Thanks for the sign-off. It's better to rebase this PR after https://github.com/aspnet/DNX/pull/1404 is merged in.

**Comment by ChengTian** [comment]
From @davidfowl : this is a good change to practice lock file versioning. We can push this PR after we have version check for lockfile

**Comment by ChengTian** [comment]
Just verified, this PR **doesn't** fix the issue #1410 . If using path separator of current OS can cause an issue, it will only happen when we `kpm restore` on one OS and move the app to another OS to run it.

Need further investigation in issue #1410

**Pull request merged**